### PR TITLE
bug with knowing user id profile

### PR DESCRIPTION
**Pull request by JMStudiosJoe**
if user trys to access other profile with another users id then will … go to login

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,12 +55,12 @@ app.get('/api/users', (req, res) => {
 });
 
 app.get('/api/user/:id', (req, res) => {
-    if (req.isAuthenticated()) {
+    if (req.isAuthenticated() && req.params.id == req.user._id) {
         db.getById('user', req.params.id).then(user => {
             return res.json({ user });
         });
     } else {
-        return res.json({ error: 'Not authenticated' });
```
Comment (JMStudiosJoe) [inline index.js]: these get caught in the frontend as successful responses for some reason

File: package.json [hunk 2]
```diff
@@ -14,8 +14,10 @@
     "deploy-windows": "SET NODE_ENV=production & webpack --config webpack.production.config.js",
     "heroku-postbuild": "webpack -p --config ./webpack.production.config.js --progress"
   },
-    "engines" : { "node" : "6" },
-    "dependencies": {
+  "engines": {
```
Comment (JMStudiosJoe) [inline package.json]: is this needed and should we update it?

Comment (naderchehab) [reply package.json]: Yes this is needed by Heroku. We should be using Node 6 and not Node 8. Some module, I forgot which one, fails with Node 8

File: index.js (hunk 1 above)
Comment (naderchehab) [reply index.js]: Maybe try `res.status(403).json({error: 'Not authenticated'});`

File: package.json (hunk 2 above)
Comment (JMStudiosJoe) [reply package.json]: got it

File: index.js (hunk 1 above)
Comment (JMStudiosJoe) [reply index.js]: I did right below it and actually was still causing the same behavior.

**Pull request merged**